### PR TITLE
TFKUBE-301: Refactor pv, pvc, namespace to used by all products

### PR DESCRIPTION
**Pull request by nghazali**

### CODE REVIEW

**Review by errcode1202 — APPROVED** [review]

File: modules/common/outputs.tf [hunk 1]
```diff
@@ -27,4 +27,14 @@ output "efs" {
...
+}
+
+output "pvc_claim_name" {
+  value       = kubernetes_persistent_volume_claim.atlassian-dc-bamboo-share-home-pvc.metadata[0].name
+  description = "Persistent volume claim name"
+}
+
+output "namespace" {
+  value       = kubernetes_namespace.products.metadata[0].name
+  description = "Namespace name for the all products"
```
Comment (nanux) [inline modules/common/outputs.tf:39]:
```suggestion
  description = "Namespace name for all products"
```

File: modules/common/variables.tf [hunk 2]
```diff
@@ -27,3 +27,17 @@ variable "domain" {
   type        = string
 }
 
+variable "namespace" {
+  description = "Namespace for the atlassian deployment."
```
Comment (nanux) [inline modules/common/variables.tf:31]:
```suggestion
  description = "Namespace for Atlassian products."
```

File: modules/products/bamboo/variables.tf [hunk 3]
```diff
@@ -12,6 +12,11 @@ variable "environment_name" {
   }
 }
 
+variable "namespace" {
+  description = "EKS namespace to install Bamboo."
```
Comment (nanux) [inline modules/products/bamboo/variables.tf:16]:
```suggestion
  description = "Kubernetes namespace to install Bamboo."
```

File: modules/products/bamboo/variables.tf [hunk 4]
```diff
@@ -12,6 +12,11 @@ variable "environment_name" {
...
+  type        = string
+}
+
 variable "vpc" {
   description = "vpc module that hosts the product."
```
Comment (nanux) [inline modules/products/bamboo/variables.tf:21]:
```suggestion
  description = "VPC module that hosts the products."
```

File: modules/products/bamboo/variables.tf [hunk 5]
```diff
@@ -116,11 +111,19 @@ variable "local_bamboo_chart_path" {
...
+  type        = string
   validation {
     condition     = can(regex("^[.?\\/?[a-zA-Z0-9|\\-|_]*]*$", var.local_agent_chart_path))
     error_message = "Invalid local Bamboo Agent Helm chart path."
   }
   default = ""
 }
 
+variable "pvc_claim_name" {
+  description = "Shared home persistent volume size."
```
Comment (nanux) [inline modules/products/bamboo/variables.tf:123]:
```suggestion
  description = "Persistent volume claim name for shared home."
```

File: modules/products/bamboo/variables.tf [hunk 6]
```diff
@@ -116,11 +111,19 @@ variable "local_bamboo_chart_path" {
...
+  type        = string
+  validation {
```
Comment (nanux) [inline modules/products/bamboo/variables.tf:125]: Is this validation just a relict from copy and paste?

File: modules/products/bamboo/variables.tf [hunk 7]
```diff
@@ -116,11 +111,19 @@ variable "local_bamboo_chart_path" {
...
+    condition     = can(regex("^[a-zA-Z]+[a-zA-Z0-9|\\-|_]*$", var.pvc_claim_name))
+    error_message = "Invalid local Bamboo Helm chart path."
```
Comment (nghazali) [inline modules/products/bamboo/variables.tf:127]:
```suggestion
    error_message = "Invalid claim name."
```

**Review by nanux — APPROVED** [review]

**Pull request merged**